TypeCobol/Compiler/Sql/Model/TruncateClause.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.IO;
+using TypeCobol.Compiler.CodeElements;
+
+namespace TypeCobol.Compiler.Sql.Model
+{
+    public class TruncateClause : SqlObject
+    {
+        public TruncateClause(SymbolReference Name)
+        {
+            TableName = new IntrinsicStorageArea(Name);
+        }
+
+        private StorageArea TableName { get; }
+        public bool IsDropStorage { get; set; }
+        public bool IsReuseStorage { get; set; }
+        public bool IsIgnoreDeleteTriggers { get; set; }
+        public bool IsRestrictWhenDeleteTriggers { get; set; }
```

Comment:
No this model is incorrect because DROP STORAGE and REUSE STORAGE are mutually exclusive, same goes for IGNORE DELETE TRIGGERS and RESTRICT WHEN DELETE TRIGGERS. I think we should go for enums and SyntaxProperties.
Also the  IMMEDIATE option is missing, you can use a `SyntaxProperty<bool>` for this one.

Remember: you should not use primitive types (string, int, bool, ...) in SqlObjects because a primitive type does not capture the syntax (tokens). It does capture semantics but not syntax.